benchmarks/benchmarks-core/src/jmh/java/io/micrometer/benchmark/core/MeterRegistrationBenchmark.java

```diff
@@ -17,44 +17,72 @@
 
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
-import org.openjdk.jmh.annotations.*;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
 import org.openjdk.jmh.runner.Runner;
 import org.openjdk.jmh.runner.RunnerException;
 import org.openjdk.jmh.runner.options.Options;
 import org.openjdk.jmh.runner.options.OptionsBuilder;
 
-import java.util.concurrent.TimeUnit;
-
-@State(Scope.Benchmark)
+@Measurement(iterations = 5, time = 1)
+@Warmup(iterations = 10, time = 1)
+@Fork(3)
+@BenchmarkMode(Mode.SampleTime)
 @OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Thread)
 public class MeterRegistrationBenchmark {
-    public static void main(String[] args) throws RunnerException {
-        Options opt = new OptionsBuilder()
-                .include(MeterRegistrationBenchmark.class.getSimpleName())
-                .warmupIterations(2)
-                .measurementIterations(5)
-                .mode(Mode.SampleTime)
-                .timeUnit(TimeUnit.SECONDS)
-                .forks(1)
-                .build();
+    MeterRegistry registry = new SimpleMeterRegistry();
+    AtomicLong i = new AtomicLong();
 
-        new Runner(opt).run();
+    @TearDown(Level.Iteration)
+    public void clear() {
+        i.set(0);
+        registry.clear();
     }
 
-    private int x = 923;
-    private int y = 123;
+    @Benchmark
+    public void register_counter() {
+        registry.counter("counter." + i.incrementAndGet(), "k", "v");
```

Comment:
I guess the registry ends up with millions of metrics, I wonder if this is representative enough of real usage since that's a really huge hashmap that probably results in most overhead.

Reply:
Yeah I'm assuming limited cardinality. I think for most monitoring systems, high cardinality will break the system, e.g., Prometheus or elasticsearch so no one should be monitoring servers that export millions of metrics and my opinion is the large map isn't very representative. I do understand that otherwise we don't have a good way of benchmarking the non-redundant case though...